### PR TITLE
chore(storage-browser): refactor description component for reusability

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

This PR consolidates usages of `<dl>`, `<dt>` and `<dd>` elements into a `DescriptionList` presentational component and migrates the `Summary` control into a `StatusDisplay` composable/stateless component and `StatusDisplay` control.

The elements were renamed from `Definition___` elements to `Description___` elements per their tag names (see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl, https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt and https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dd)

#### Description of how you validated changes

* Tested components locally in sample app
* Updated/added unit tests
* `yarn test`

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [x] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by AllanZhengYP — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/displayText/en.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Temporarily just export strings until displayText strategy is finalized
+ */
+export const displayText = {
+  uploadDestination: 'Destination',
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/displayText/en.ts:5]: Tangential to the actual PR: can genericize the key here, "destination" will be used for other actions (e.g. `copy`)

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx [hunk 2]
```diff
@@ -196,6 +182,8 @@ export const UploadControls = (): JSX.Element => {
     { prefix: path, preventOverwrite }
   );
 
+  // Replace with stuff
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx:185]: Depends on the stuff 😅

**Review by calebpollman — DISMISSED** [review]
🚢

File: packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx (hunk 2 above)
Comment (cshfang) [reply packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx:185]: Oops I have no idea what stuff this was even meant to be now. I'll just remove the comment

File: packages/react-storage/src/components/StorageBrowser/displayText/en.ts (hunk 1 above)
Comment (cshfang) [reply packages/react-storage/src/components/StorageBrowser/displayText/en.ts:5]: I'll change it to actionDestination so it's not overly broad

**Review by calebpollman — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**